src/runtime_src/hip/core/event.cpp

```diff
@@ -80,19 +77,14 @@ std::shared_ptr<stream> event::get_stream()
 
 void event::add_to_chain(std::shared_ptr<command> cmd)
 {
-  // lock and add
+  std::lock_guard<std::mutex> lock(m_mutex);
+  chain_of_commands.push_back(cmd);
 }
 
 void event::add_dependency(std::shared_ptr<command> cmd)
 {
-  // lock and add
-}
-
-float event::elapsed_time (const std::shared_ptr<command>& end)
-{
-  auto duration = end->get_time() - get_time();
-  auto millis = std::chrono::duration_cast<std::chrono::milliseconds>(duration).count();
-  return millis;
+  std::lock_guard<std::mutex> lock(m_mutex);
+  recorded_commands.push_back(cmd);
```

Comment:
pass by value means you should should preferably std::move(cmd) here to turn cmd into an rvalue.